common/workunit/workunit.cpp

```diff
@@ -874,6 +884,106 @@ class CDaliWuGraphStats : public CWuGraphStats
     Owned<IRemoteConnection> conn;
 };
 
+CWorkUnitWatcher::CWorkUnitWatcher(IWorkUnitSubscriber *_subscriber, WUSubscribeOptions flags, const char *wuid) : subscriber(_subscriber)
+{
+    abort = 0;
+    state = 0;
+    action = 0;
+    assertex((flags & ~SubscribeOptionAbort) == 0);
+    if (flags & SubscribeOptionAbort)
+    {
+        VStringBuffer xpath("/WorkUnitAborts/%s", wuid);
+        abort = querySDS().subscribe(xpath.str(), *this);
+    }
+}
+CWorkUnitWatcher::~CWorkUnitWatcher()
+{
+    assertex(abort==0 && state==0 && action==0);
+}
+
+void CWorkUnitWatcher::unsubscribe()
+{
+    CriticalBlock b(crit);
+    if (abort)
+        querySDS().unsubscribe(abort);
+    if (state)
+        querySDS().unsubscribe(state);
+    if (action)
+        querySDS().unsubscribe(action);
+    abort = 0;
+    state = 0;
+    action = 0;
```

Comment:
wondering if this method (unsubscribe()) shouldn't make be called in the dtor of CWorkUnitWatch.. to ensure subscription isn't hanging around to dead objects.